### ISSUE TITLE
Date Unknown QuickTime (Apple) MOV/MP4 videos

### ISSUE THREAD

**Issue by uxusoft** [photoprism/photoprism#727]
Hello!

When I import mov files, for example:

> # exiftool 20150211_191910_062ACF0D.mov
> ExifTool Version Number         : 11.16
> File Name                       : 20150211_191910_062ACF0D.mov
> Directory                       : .
> File Size                       : 349 MB
> File Modification Date/Time     : 2020:12:21 15:21:18+01:00
> File Access Date/Time           : 2020:12:21 15:21:18+01:00
> File Inode Change Date/Time     : 2020:12:21 15:21:18+01:00
> File Permissions                : rwxr-xr-x
> File Type                       : MOV
> File Type Extension             : mov
> MIME Type                       : video/quicktime
> Major Brand                     : Apple QuickTime (.MOV/QT)
> Minor Version                   : 0.0.0
> Compatible Brands               : qt
> Movie Data Size                 : 366267618
> Movie Data Offset               : 36
> Movie Header Version            : 0
> Time Scale                      : 600
> Duration                        : 0:02:50
> Preferred Rate                  : 1
> Preferred Volume                : 100.00%
> Preview Time                    : 0 s
> Preview Duration                : 0 s
> Poster Time                     : 0 s
> Selection Time                  : 0 s
> Selection Duration              : 0 s
> Current Time                    : 0 s
> Next Track ID                   : 3
> Track Header Version            : 0
> Track Create Date               : 2015:02:11 19:19:10
> Track Modify Date               : 2015:02:11 19:22:00
> Track ID                        : 1
> Track Duration                  : 0:02:50
> Track Layer                     : 0
> Track Volume                    : 0.00%
> Image Width                     : 1920
> Image Height                    : 1080
> Clean Aperture Dimensions       : 1920x1080
> Production Aperture Dimensions  : 1920x1080
> Encoded Pixels Dimensions       : 1920x1080
> Graphics Mode                   : ditherCopy
> Op Color                        : 32768 32768 32768
> Compressor ID                   : avc1
> Source Image Width              : 1920
> Source Image Height             : 1080
> X Resolution                    : 72
> Y Resolution                    : 72
> Compressor Name                 : H.264
> Bit Depth                       : 24
> Video Frame Rate                : 30.006
> Matrix Structure                : 1 0 0 0 1 0 0 0 1
> Media Header Version            : 0
> Media Create Date               : 2015:02:11 19:19:10
> Media Modify Date               : 2015:02:11 19:22:00
> Media Time Scale                : 44100
> Media Duration                  : 0:02:50
> Media Language Code             : und
> Balance                         : 0
> Handler Class                   : Data Handler
> Handler Vendor ID               : Apple
> Handler Description             : Core Media Data Handler
> Audio Format                    : mp4a
> Audio Bits Per Sample           : 16
> Audio Sample Rate               : 44100
> Layout Flags                    : Mono
> Audio Channels                  : 1
> Purchase File Format            : mp4a
> Handler Type                    : Metadata Tags
> Creation Date (rus-RU)          : 2015:02:11 22:19:10+03:00
> Model (rus-RU)                  : iPhone 6 Plus
> Software (rus-RU)               : 8.1.3
> GPS Coordinates (rus-RU)        : 56 deg 15' 7.92" N, 37 deg 26' 21.48" E, 181.101 m Above Sea Level
> Make (rus-RU)                   : Apple
> Software Version (rus)          : 8.1.3
> Content Create Date (rus)       : 2015:02:11 22:19:10+03:00
> GPS Coordinates (rus)           : 56 deg 15' 7.92" N, 37 deg 26' 21.48" E, 181.101 m Above Sea Level
> Model (rus)                     : iPhone 6 Plus
> Make (rus)                      : Apple
> XMP Toolkit                     : Image::ExifTool 11.16
> **Date/Time Original              : 2015:02:11 22:19:10+03:00**
> Create Date                     : 2015:02:11 22:19:10+03:00
> Modify Date                     : 2015:02:11 22:19:10+03:00
> Creation Date                   : 2015:02:11 22:19:10+03:00
> Model                           : iPhone 6 Plus
> Software                        : 8.1.3
> GPS Coordinates                 : 56 deg 15' 7.92" N, 37 deg 26' 21.48" E, 181.101 m Above Sea Level
> Make                            : Apple
> Software Version                : 8.1.3
> Content Create Date             : 2015:02:11 22:19:10+03:00
> Avg Bitrate                     : 17.3 Mbps
> GPS Altitude                    : 181.101 m
> GPS Altitude Ref                : Above Sea Level
> GPS Latitude                    : 56 deg 15' 7.92" N
> GPS Longitude                   : 37 deg 26' 21.48" E
> Image Size                      : 1920x1080
> Megapixels                      : 2.1
> Rotation                        : 0
> GPS Position                    : 56 deg 15' 7.92" N, 37 deg 26' 21.48" E

it is imported in Photoprism gallery with Unknown date and with Local time = UTC import time on server:
`# date
Mon Dec 21 15:27:38 CET 2020` (UTC+1:00)

<img width="404" alt="Screenshot 2020-12-21 at 15 28 28" src="https://user-images.githubusercontent.com/64733669/102787376-47e64100-43a1-11eb-9bb0-b2a6192de982.png">

**Comment by uxusoft** [comment]
Debug example:
time="2020-12-21T15:24:14Z" level=debug msg="config: loaded settings from /photoprism/storage/config/settings.yml "
time="2020-12-21T15:24:14Z" level=info msg="moving media files from /photoprism/import to /photoprism/originals"
time="2020-12-21T15:24:14Z" level=info msg="classify: loading nasnet"
time="2020-12-21T15:24:17Z" level=info msg="classify: loading labels from labels.txt"
time="2020-12-21T15:24:17Z" level=info msg="import: no .ppignore file found"
time="2020-12-21T15:24:17Z" level=info msg="import: added folder /"
time="2020-12-21T15:24:17Z" level=debug msg="media: no json sidecar file found for 20200912_205148_7A065C1C.mp4"
time="2020-12-21T15:24:17Z" level=debug msg="media: exif not supported: 20200912_205148_7A065C1C.mp4"
time="2020-12-21T15:24:17Z" level=info msg="**media: 20200912_205148_7A065C1C.mp4 was taken at 2020-09-12 16:36:10 +0000 UTC (file mod time)**"
time="2020-12-21T15:24:17Z" level=info msg="import: moving main mp4 file 20200912_205148_7A065C1C.mp4 to 2020/09/20200912_163610_DB7A2ADC.mp4"
time="2020-12-21T15:24:17Z" level=debug msg="failed renaming file, fallback to copy and delete: rename /photoprism/import/20200912_205148_7A065C1C.mp4 /photoprism/originals/2020/09/20200912_163610_DB7A2ADC.mp4: invalid cross-device link"
time="2020-12-21T15:24:17Z" level=debug msg="convert: 2020/09/20200912_163610_DB7A2ADC.mp4 -> 20200912_163610_DB7A2ADC.mp4.jpg"
time="2020-12-21T15:24:17Z" level=debug msg="import: 20200912_163610_DB7A2ADC.mp4.jpg created"
time="2020-12-21T15:24:17Z" level=info msg="media: 13 thumbnails created for 20200912_163610_DB7A2ADC [268.617017ms]"
time="2020-12-21T15:24:17Z" level=debug msg="convert: 2020/09/20200912_163610_DB7A2ADC.mp4 -> 20200912_163610_DB7A2ADC.mp4.json"
time="2020-12-21T15:24:18Z" level=debug msg="import: 20200912_163610_DB7A2ADC.mp4.json created"
time="2020-12-21T15:24:18Z" level=info msg="import: added main mp4 file 2020/09/20200912_163610_DB7A2ADC.mp4"
time="2020-12-21T15:24:18Z" level=debug msg="convert: /photoprism/storage/sidecar/2020/09/20200912_163610_DB7A2ADC.mp4.jpg -> 20200912_163610_DB7A2ADC.mp4.jpg.json"
time="2020-12-21T15:24:18Z" level=debug msg="import: 20200912_163610_DB7A2ADC.mp4.jpg.json created"
time="2020-12-21T15:24:18Z" level=debug msg="metadata: failed parsing 20200912_163610_DB7A2ADC.mp4.jpg, starting brute-force search (parse jpeg)"
time="2020-12-21T15:24:20Z" level=debug msg="index: image classification took 2.276979628s"
time="2020-12-21T15:24:20Z" level=info msg=**"media: 20200912_163610_DB7A2ADC.mp4.jpg was taken at 2020-12-21 15:24:17.621060265 +0000 UTC (file mod time)"**
time="2020-12-21T15:24:20Z" level=debug msg="photo: changed title of pqlp44i27uzska3n to Moment"
time="2020-12-21T15:24:20Z" level=debug msg="index: updated yaml file 20200912_163610_DB7A2ADC.yml"
time="2020-12-21T15:24:20Z" level=debug msg="media: no new thumbnails created for 20200912_163610_DB7A2ADC [111.275µs]"
time="2020-12-21T15:24:20Z" level=info msg="import: added related jpg file /photoprism/storage/sidecar/2020/09/20200912_163610_DB7A2ADC.mp4.jpg"
time="2020-12-21T15:24:20Z" level=info msg="import completed in 5.97032534s"
time="2020-12-21T15:24:20Z" level=info msg="closed database connection"

**Comment by lastzero** [comment]
Thanks for reporting this! Can you provide us with the ExifTool output as JSON and (optionally) the original file for testing?

```
exiftool -j example.jpg > example.json
```

**Comment by uxusoft** [comment]
File for test:
[https://test-zxb.s3.eu-central-1.amazonaws.com/20150610_110609_663B50DC.mov](url)

ExifTool JSON output:
```
[{
  "SourceFile": "20150610_110609_663B50DC.mov",
  "ExifToolVersion": 12.00,
  "FileName": "20150610_110609_663B50DC.mov",
  "Directory": ".",
  "FileSize": "14 MB",
  "FileModifyDate": "2020:12:21 15:42:30+01:00",
  "FileAccessDate": "2020:12:21 22:28:48+01:00",
  "FileInodeChangeDate": "2020:12:21 22:28:30+01:00",
  "FilePermissions": "rwxr-xr-x",
  "FileType": "MOV",
  "FileTypeExtension": "mov",
  "MIMEType": "video/quicktime",
  "MajorBrand": "Apple QuickTime (.MOV/QT)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["qt  "],
  "MediaDataSize": 14675292,
  "MediaDataOffset": 36,
  "MovieHeaderVersion": 0,
  "TimeScale": 600,
  "Duration": "6.83 s",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 3,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2015:06:10 11:06:09",
  "TrackModifyDate": "2015:06:10 11:06:16",
  "TrackID": 1,
  "TrackDuration": "6.83 s",
  "TrackLayer": 0,
  "TrackVolume": "0.00%",
  "ImageWidth": 1920,
  "ImageHeight": 1080,
  "CleanApertureDimensions": "1920x1080",
  "ProductionApertureDimensions": "1920x1080",
  "EncodedPixelsDimensions": "1920x1080",
  "GraphicsMode": "ditherCopy",
  "OpColor": "32768 32768 32768",
  "CompressorID": "avc1",
  "SourceImageWidth": 1920,
  "SourceImageHeight": 1080,
  "XResolution": 72,
  "YResolution": 72,
  "CompressorName": "H.264",
  "BitDepth": 24,
  "VideoFrameRate": 30.007,
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2015:06:10 11:06:09",
  "MediaModifyDate": "2015:06:10 11:06:16",
  "MediaTimeScale": 44100,
  "MediaDuration": "6.90 s",
  "MediaLanguageCode": "und",
  "Balance": 0,
  "HandlerClass": "Data Handler",
  "HandlerVendorID": "Apple",
  "HandlerDescription": "Core Media Data Handler",
  "AudioFormat": "mp4a",
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "LayoutFlags": "Mono",
  "AudioChannels": 1,
  "PurchaseFileFormat": "mp4a",
  "SoftwareVersion-rus": 8.3,
  "ContentCreateDate-rus": "2015:06:10 14:06:09+03:00",
  "GPSCoordinates-rus": "55 deg 33' 48.96\" N, 37 deg 58' 56.64\" E, 141.315 m Above Sea Level",
  "Model-rus": "iPhone 6 Plus",
  "Make-rus": "Apple",
  "XMPToolkit": "Image::ExifTool 11.16",
  "DateTimeOriginal": "2015:06:10 14:06:09+03:00",
  "CreateDate": "2015:06:10 14:06:09+03:00",
  "ModifyDate": "2015:06:10 14:06:09+03:00",
  "HandlerType": "Metadata Tags",
  "CreationDate-rus-RU": "2015:06:10 14:06:09+03:00",
  "Model-rus-RU": "iPhone 6 Plus",
  "Software-rus-RU": 8.3,
  "GPSCoordinates-rus-RU": "55 deg 33' 48.96\" N, 37 deg 58' 56.64\" E, 141.315 m Above Sea Level",
  "Make-rus-RU": "Apple",
  "SoftwareVersion": 8.3,
  "ContentCreateDate": "2015:06:10 14:06:09+03:00",
  "GPSCoordinates": "55 deg 33' 48.96\" N, 37 deg 58' 56.64\" E, 141.315 m Above Sea Level",
  "Model": "iPhone 6 Plus",
  "Make": "Apple",
  "CreationDate": "2015:06:10 14:06:09+03:00",
  "Software": 8.3,
  "ImageSize": "1920x1080",
  "Megapixels": 2.1,
  "AvgBitrate": "17.2 Mbps",
  "GPSAltitude": "141.315 m",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSLatitude": "55 deg 33' 48.96\" N",
  "GPSLongitude": "37 deg 58' 56.64\" E",
  "Rotation": 90,
  "GPSPosition": "55 deg 33' 48.96\" N, 37 deg 58' 56.64\" E"
}]
```

Demo website with this file imported:
[http://18.193.242.88:2342/videos](url)

**Comment by lastzero** [comment]
Seems ExifTool or Apple sometimes add time zones to create dates. Never seen this before, should work now. Please test!

**Comment by lastzero** [comment]
![Screenshot 2020-12-22 at 02 03 02](https://user-images.githubusercontent.com/301686/102836516-dedee780-43f9-11eb-92e8-338b68a693df.png)

**Comment by uxusoft** [comment]
Thank you very much for your efforts!

I tested with original file from iPhone 7: [IMG_6889.MOV](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_6889.MOV).

Exif JSON:
```
[{
  "SourceFile": "IMG_6889.MOV",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_6889.MOV",
  "Directory": ".",
  "FileSize": "26 MB",
  "FileModifyDate": "2020:12:20 00:14:32+01:00",
  "FileAccessDate": "2020:12:22 02:15:22+01:00",
  "FileInodeChangeDate": "2020:12:22 02:07:04+01:00",
  "FilePermissions": "rw-r--r--",
  "FileType": "MOV",
  "FileTypeExtension": "mov",
  "MIMEType": "video/quicktime",
  "MajorBrand": "Apple QuickTime (.MOV/QT)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["qt  "],
  "MediaDataSize": 26754461,
  "MediaDataOffset": 36,
  "MovieHeaderVersion": 0,
  "CreateDate": "2020:12:19 23:14:27",
  "ModifyDate": "2020:12:19 23:14:32",
  "TimeScale": 600,
  "Duration": "28.60 s",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 3,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2020:12:19 23:14:27",
  "TrackModifyDate": "2020:12:19 23:14:32",
  "TrackID": 1,
  "TrackDuration": "28.53 s",
  "TrackLayer": 0,
  "TrackVolume": "100.00%",
  "Balance": 0,
  "AudioFormat": "mp4a",
  "AudioChannels": 1,
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "PurchaseFileFormat": "mp4a",
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "ImageWidth": 1280,
  "ImageHeight": 720,
  "CleanApertureDimensions": "1280x720",
  "ProductionApertureDimensions": "1280x720",
  "EncodedPixelsDimensions": "1280x720",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2020:12:19 23:14:27",
  "MediaModifyDate": "2020:12:19 23:14:32",
  "MediaTimeScale": 600,
  "MediaDuration": "28.60 s",
  "MediaLanguageCode": "und",
  "GraphicsMode": "ditherCopy",
  "OpColor": "32768 32768 32768",
  "HandlerClass": "Data Handler",
  "HandlerVendorID": "Apple",
  "HandlerDescription": "Core Media Data Handler",
  "CompressorID": "avc1",
  "SourceImageWidth": 1280,
  "SourceImageHeight": 720,
  "XResolution": 72,
  "YResolution": 72,
  "CompressorName": "H.264",
  "BitDepth": 24,
  "VideoFrameRate": 30,
  "HandlerType": "Metadata Tags",
  "GPSCoordinates": "36 deg 7' 52.68\" N, 51 deg 42' 13.32\" E, 11100.727 m Above Sea Level",
  "Make": "Apple",
  "Model": "iPhone 7",
  "Software": "10.2.1",
  **"CreationDate": "2017:03:27 19:28:47+04:30",**
  "ImageSize": "1280x720",
  "Megapixels": 0.922,
  "AvgBitrate": "7.48 Mbps",
  "GPSAltitude": "11100.727 m",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSLatitude": "36 deg 7' 52.68\" N",
  "GPSLongitude": "51 deg 42' 13.32\" E",
  "Rotation": 0,
  "GPSPosition": "36 deg 7' 52.68\" N, 51 deg 42' 13.32\" E"
}]
```
Imported to [sandbox](http://18.193.242.88:2342/videos):

<img width="672" alt="Screenshot 2020-12-22 at 02 19 57" src="https://user-images.githubusercontent.com/64733669/102837542-485ff580-43fc-11eb-8426-54d233986dbd.png">

Photoprism now displays date! Maybe it is correct to take date from CreationDate because it contains correct timezone.

To be sure, I test with another iPhone 6 Plus file [IMG_2699.MOV](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_2699.MOV)

Exif JSON:
```
[{
  "SourceFile": "IMG_2699.MOV",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_2699.MOV",
  "Directory": ".",
  "FileSize": "59 MB",
  "FileModifyDate": "2020:07:16 22:53:34+02:00",
  "FileAccessDate": "2020:12:22 02:25:01+01:00",
  "FileInodeChangeDate": "2020:12:22 02:07:04+01:00",
  "FilePermissions": "rw-------",
  "FileType": "MOV",
  "FileTypeExtension": "mov",
  "MIMEType": "video/quicktime",
  "MajorBrand": "Apple QuickTime (.MOV/QT)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["qt  "],
  "MediaDataSize": 61433770,
  "MediaDataOffset": 36,
  "MovieHeaderVersion": 0,
  "CreateDate": "2015:04:28 05:28:48",
  "ModifyDate": "2015:04:28 05:29:16",
  "TimeScale": 600,
  "Duration": "27.83 s",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 3,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2015:04:28 05:28:48",
  "TrackModifyDate": "2015:04:28 05:29:16",
  "TrackID": 1,
  "TrackDuration": "27.83 s",
  "TrackLayer": 0,
  "TrackVolume": "0.00%",
  "ImageWidth": 1920,
  "ImageHeight": 1080,
  "CleanApertureDimensions": "1920x1080",
  "ProductionApertureDimensions": "1920x1080",
  "EncodedPixelsDimensions": "1920x1080",
  "GraphicsMode": "ditherCopy",
  "OpColor": "32768 32768 32768",
  "CompressorID": "avc1",
  "SourceImageWidth": 1920,
  "SourceImageHeight": 1080,
  "XResolution": 72,
  "YResolution": 72,
  "CompressorName": "H.264",
  "BitDepth": 24,
  "VideoFrameRate": 30.005,
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2015:04:28 05:28:48",
  "MediaModifyDate": "2015:04:28 05:29:16",
  "MediaTimeScale": 44100,
  "MediaDuration": "27.89 s",
  "MediaLanguageCode": "und",
  "Balance": 0,
  "HandlerClass": "Data Handler",
  "HandlerVendorID": "Apple",
  "HandlerDescription": "Core Media Data Handler",
  "AudioFormat": "mp4a",
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "LayoutFlags": "Mono",
  "AudioChannels": 1,
  "PurchaseFileFormat": "mp4a",
  "HandlerType": "Metadata Tags",
  "CreationDate-rus-RU": "2015:04:28 10:28:48+05:00",
  "Model-rus-RU": "iPhone 6 Plus",
  "Software-rus-RU": 8.3,
  "GPSCoordinates-rus-RU": "54 deg 57' 10.08\" N, 61 deg 30' 14.04\" E, 236.374 m Above Sea Level",
  "Make-rus-RU": "Apple",
  "SoftwareVersion-rus": 8.3,
  "ContentCreateDate-rus": "2015:04:28 10:28:48+05:00",
  "GPSCoordinates-rus": "54 deg 57' 10.08\" N, 61 deg 30' 14.04\" E, 236.374 m Above Sea Level",
  "Model-rus": "iPhone 6 Plus",
  "Make-rus": "Apple",
  "CreationDate": "2015:04:28 10:28:48+05:00",
  "Model": "iPhone 6 Plus",
  "Software": 8.3,
  "GPSCoordinates": "54 deg 57' 10.08\" N, 61 deg 30' 14.04\" E, 236.374 m Above Sea Level",
  "Make": "Apple",
  "SoftwareVersion": 8.3,
  "ContentCreateDate": "2015:04:28 10:28:48+05:00",
  "ImageSize": "1920x1080",
  "Megapixels": 2.1,
  "AvgBitrate": "17.7 Mbps",
  "GPSAltitude": "236.374 m",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSLatitude": "54 deg 57' 10.08\" N",
  "GPSLongitude": "61 deg 30' 14.04\" E",
  "Rotation": 0,
  "GPSPosition": "54 deg 57' 10.08\" N, 61 deg 30' 14.04\" E"
}]
```

Imported to [sandbox](http://18.193.242.88:2342/videos):
<img width="400" alt="Screenshot 2020-12-22 at 02 29 33" src="https://user-images.githubusercontent.com/64733669/102838080-93c6d380-43fd-11eb-9f5e-5061d5a3fcc3.png">

Time is incorrect here. Correct time is Local 15:28 UTC 10:28, but Photoprism shows 05:28 Local and 00:28 UTC.

**Comment by lastzero** [comment]
Well, the timestamp in the JSON file is 2015:04:28 05:28:48, that's what we use - unless you go for the non standard "-rus" date field. We can take a look at this another time.

**Comment by uxusoft** [comment]
Thank you very much!
Sorry but correct time is in "CreationDate" field. For example, there are no "rus" fields, just another original file from iPhone 6 Plus:

[IMG_4561.MOV](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_4561.MOV)

```
[{
  "SourceFile": "IMG_4561.MOV",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_4561.MOV",
  "Directory": ".",
  "FileSize": "22 MB",
  "FileModifyDate": "2020:07:16 19:06:26+02:00",
  "FileAccessDate": "2020:12:22 02:37:08+01:00",
  "FileInodeChangeDate": "2020:12:22 02:07:04+01:00",
  "FilePermissions": "rw-------",
  "FileType": "MOV",
  "FileTypeExtension": "mov",
  "MIMEType": "video/quicktime",
  "MajorBrand": "Apple QuickTime (.MOV/QT)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["qt  "],
  "MediaDataSize": 23138648,
  "MediaDataOffset": 36,
  "MovieHeaderVersion": 0,
  "CreateDate": "2015:12:06 15:22:29",
  "ModifyDate": "2015:12:06 15:22:40",
  "TimeScale": 600,
  "Duration": "10.67 s",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 5,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2015:12:06 15:22:29",
  "TrackModifyDate": "2015:12:06 15:22:40",
  "TrackID": 1,
  "TrackDuration": "10.67 s",
  "TrackLayer": 0,
  "TrackVolume": "100.00%",
  "ImageWidth": 1920,
  "ImageHeight": 1080,
  "CleanApertureDimensions": "1920x1080",
  "ProductionApertureDimensions": "1920x1080",
  "EncodedPixelsDimensions": "1920x1080",
  "GraphicsMode": "ditherCopy",
  "OpColor": "32768 32768 32768",
  "CompressorID": "avc1",
  "SourceImageWidth": 1920,
  "SourceImageHeight": 1080,
  "XResolution": 72,
  "YResolution": 72,
  "CompressorName": "H.264",
  "BitDepth": 24,
  "VideoFrameRate": 29.981,
  "Balance": 0,
  "AudioFormat": "mp4a",
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "LayoutFlags": "Mono",
  "AudioChannels": 1,
  "PurchaseFileFormat": "mp4a",
  "Warning": "[minor] The ExtractEmbedded option may find more tags in the media data",
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "ContentDescribes": "Track 1",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2015:12:06 15:22:29",
  "MediaModifyDate": "2015:12:06 15:22:40",
  "MediaTimeScale": 600,
  "MediaDuration": "10.67 s",
  "MediaLanguageCode": "und",
  "GenMediaVersion": 0,
  "GenFlags": "0 0 0",
  "GenGraphicsMode": "ditherCopy",
  "GenOpColor": "32768 32768 32768",
  "GenBalance": 0,
  "HandlerClass": "Data Handler",
  "HandlerVendorID": "Apple",
  "HandlerDescription": "Core Media Data Handler",
  "MetaFormat": "mebx",
  "HandlerType": "Metadata Tags",
  "GPSCoordinates": "55 deg 45' 28.44\" N, 37 deg 37' 10.92\" E, 135.89 m Above Sea Level",
  "Make": "Apple",
  "Model": "iPhone 6 Plus",
  "Software": 9.1,
  "CreationDate": "2015:12:06 18:22:29+03:00",
  "ImageSize": "1920x1080",
  "Megapixels": 2.1,
  "AvgBitrate": "17.3 Mbps",
  "GPSAltitude": "135.89 m",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSLatitude": "55 deg 45' 28.44\" N",
  "GPSLongitude": "37 deg 37' 10.92\" E",
  "Rotation": 0,
  "GPSPosition": "55 deg 45' 28.44\" N, 37 deg 37' 10.92\" E"
}]
```
<img width="383" alt="Screenshot 2020-12-22 at 02 41 54" src="https://user-images.githubusercontent.com/64733669/102838777-4cd9dd80-43ff-11eb-96b4-58d2c50c7571.png">

**Comment by lastzero** [comment]
It's probably because we assume timestamps are local, but in this case CreationDate has a time zone so CreateDate seems to be UTC.

**Comment by uxusoft** [comment]
To be 1000% sure, I made video now with this thread with iPhone 8 :-)

[IMG_6884.MOV](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_6884.MOV)

```
[{
  "SourceFile": "IMG_6884.MOV",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_6884.MOV",
  "Directory": ".",
  "FileSize": "9.6 MB",
  "FileModifyDate": "2020:12:22 02:46:08+01:00",
  "FileAccessDate": "2020:12:22 02:48:31+01:00",
  "FileInodeChangeDate": "2020:12:22 02:46:24+01:00",
  "FilePermissions": "rw-r--r--",
  "FileType": "MOV",
  "FileTypeExtension": "mov",
  "MIMEType": "video/quicktime",
  "MajorBrand": "Apple QuickTime (.MOV/QT)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["qt  "],
  "MediaDataSize": 10091927,
  "MediaDataOffset": 36,
  "MovieHeaderVersion": 0,
  "CreateDate": "2020:12:22 01:46:08",
  "ModifyDate": "2020:12:22 01:46:08",
  "TimeScale": 600,
  "Duration": "6.83 s",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 5,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2020:12:22 01:46:08",
  "TrackModifyDate": "2020:12:22 01:46:08",
  "TrackID": 1,
  "TrackDuration": "6.83 s",
  "TrackLayer": 0,
  "TrackVolume": "0.00%",
  "ImageWidth": 1920,
  "ImageHeight": 1080,
  "CleanApertureDimensions": "1920x1080",
  "ProductionApertureDimensions": "1920x1080",
  "EncodedPixelsDimensions": "1920x1080",
  "GraphicsMode": "ditherCopy",
  "OpColor": "32768 32768 32768",
  "CompressorID": "hvc1",
  "SourceImageWidth": 1920,
  "SourceImageHeight": 1080,
  "XResolution": 72,
  "YResolution": 72,
  "CompressorName": "HEVC",
  "BitDepth": 24,
  "VideoFrameRate": 60,
  "Balance": 0,
  "AudioFormat": "mp4a",
  "AudioChannels": 1,
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "PurchaseFileFormat": "mp4a",
  "Warning": "[minor] The ExtractEmbedded option may find more tags in the media data",
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "ContentDescribes": "Track 1",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2020:12:22 01:46:08",
  "MediaModifyDate": "2020:12:22 01:46:08",
  "MediaTimeScale": 600,
  "MediaDuration": "6.83 s",
  "MediaLanguageCode": "und",
  "GenMediaVersion": 0,
  "GenFlags": "0 0 0",
  "GenGraphicsMode": "ditherCopy",
  "GenOpColor": "32768 32768 32768",
  "GenBalance": 0,
  "HandlerClass": "Data Handler",
  "HandlerVendorID": "Apple",
  "HandlerDescription": "Core Media Data Handler",
  "MetaFormat": "mebx",
  "HandlerType": "Metadata Tags",
  "LocationAccuracyHorizontal": 65.000000,
  "Make": "Apple",
  "Model": "iPhone 8",
  "Software": 14.3,
  "CreationDate": "2020:12:22 02:45:43+01:00",
  "ImageSize": "1920x1080",
  "Megapixels": 2.1,
  "AvgBitrate": "11.8 Mbps",
  "Rotation": 90
}]
```

<img width="383" alt="Screenshot 2020-12-22 at 02 51 14" src="https://user-images.githubusercontent.com/64733669/102839341-9676f800-4400-11eb-9af7-aea80c1a3cd1.png">

% date -u
Tue Dec 22 01:51:54 UTC 2020
% date
Tue Dec 22 02:52:33 CET 2020

UTC time is correct, Local Time is not correct.

**Comment by uxusoft** [comment]
It seems that correct time = MediaCreateDate + TimeZone_from_CreationDate

**Comment by uxusoft** [comment]
MediaCreateDate is in UTC actually!
But Photoprism treats it as local time.

Correct algorithm is to treat it as UTC time and take TimeZone from CreationDate for Local Time (or location data).

**Comment by lastzero** [comment]
Files need GPS coordinates for time zone offsets.

**Comment by uxusoft** [comment]
To import time correctly, we have to treat MediaCreateDate = UTC.

Just as feature request, it could be better to first take TimeZone from CreationDate, and if it is not found - to take it from location.

**Comment by lastzero** [comment]
Beyond doubt, these are the local timestamps from your JSON files:

![gps](https://user-images.githubusercontent.com/301686/102858287-490f8080-442a-11eb-87e2-83f5c0ad6323.jpg)

![no-gps](https://user-images.githubusercontent.com/301686/102858289-4a40ad80-442a-11eb-8c91-597790d18bbe.jpg)

Note that it's not possible to do a "reverse lookup" of the actual time zone from the offset provided in the timestamp if there are no GPS coordinates. PhotoPrism uses the local time in this case.

**Comment by uxusoft** [comment]
Wow! I checked another file with GPS location. Cool, now looks like it is importing UTC time correctly!
Thank you very much for bugfix!

> PhotoPrism uses the local time in this case.
Could you please clarify what do you mean? It looks like if there are no GPS in file, PhotoPrism makes Local Time = MediaCreateDate.

For local time, it could be good to add (sometime later) feature:
if exif_json(CreationDate) contain TimeZone like XXX+XX:XX AND Make like "Apple":
Local Time = MediaCreateDate + TimeZone
and only after that to look for GPS location

**Comment by uxusoft** [comment]
Also I checked with [photo from Berlin IMG_0025](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_0025.JPG)

```
[{
  "SourceFile": "IMG_0025.JPG",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_0025.JPG",
  "Directory": ".",
  "FileSize": "2027 kB",
  "FileModifyDate": "2020:07:16 22:58:06+02:00",
  "FileAccessDate": "2020:12:22 02:07:37+01:00",
  "FileInodeChangeDate": "2020:12:22 02:07:04+01:00",
  "FilePermissions": "rw-------",
  "FileType": "JPEG",
  "FileTypeExtension": "jpg",
  "MIMEType": "image/jpeg",
  "ExifByteOrder": "Big-endian (Motorola, MM)",
  "Make": "Apple",
  "Model": "iPhone 6 Plus",
  "Orientation": "Horizontal (normal)",
  "XResolution": 72,
  "YResolution": 72,
  "ResolutionUnit": "inches",
  "Software": 8.2,
  "ModifyDate": "2015:03:28 23:45:05",
  "YCbCrPositioning": "Centered",
  "ExposureTime": "1/15",
  "FNumber": 2.2,
  "ExposureProgram": "Program AE",
  "ISO": 40,
  "ExifVersion": "0221",
  "DateTimeOriginal": "2015:03:28 23:45:05",
  "CreateDate": "2015:03:28 23:45:05",
  "ComponentsConfiguration": "Y, Cb, Cr, -",
  "ShutterSpeedValue": "1/15",
  "ApertureValue": 2.2,
  "BrightnessValue": 2.236479678,
  "ExposureCompensation": "+0.00391",
  "MeteringMode": "Multi-segment",
  "Flash": "Auto, Did not fire",
  "FocalLength": "4.2 mm",
  "SubjectArea": "1631 1223 1795 1077",
  "RunTimeFlags": "Valid",
  "RunTimeValue": 222808582383250,
  "RunTimeEpoch": 0,
  "RunTimeScale": 1000000000,
  "AccelerationVector": "-0.9963754984 -0.01370004831 -0.06765389082",
  "SubSecTimeOriginal": 967,
  "SubSecTimeDigitized": 967,
  "FlashpixVersion": "0100",
  "ColorSpace": "sRGB",
  "ExifImageWidth": 3264,
  "ExifImageHeight": 2448,
  "SensingMethod": "One-chip color area",
  "SceneType": "Directly photographed",
  "ExposureMode": "Auto",
  "WhiteBalance": "Auto",
  "FocalLengthIn35mmFormat": "29 mm",
  "SceneCaptureType": "Standard",
  "LensInfo": "4.15mm f/2.2",
  "LensMake": "Apple",
  "LensModel": "iPhone 6 Plus back camera 4.15mm f/2.2",
  "GPSLatitudeRef": "North",
  "GPSLongitudeRef": "East",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSTimeStamp": "20:44:45",
  "GPSSpeedRef": "km/h",
  "GPSSpeed": 0,
  "GPSDateStamp": "2015:03:28",
  "Compression": "JPEG (old-style)",
  "ThumbnailOffset": 1996,
  "ThumbnailLength": 10896,
  "ImageWidth": 3264,
  "ImageHeight": 2448,
  "EncodingProcess": "Baseline DCT, Huffman coding",
  "BitsPerSample": 8,
  "ColorComponents": 3,
  "YCbCrSubSampling": "YCbCr4:2:0 (2 2)",
  "RunTimeSincePowerUp": "2 days 13:53:29",
  "Aperture": 2.2,
  "ImageSize": "3264x2448",
  "LensID": "iPhone 6 Plus back camera 4.15mm f/2.2",
  "Megapixels": 8.0,
  "ScaleFactor35efl": 7.0,
  "ShutterSpeed": "1/15",
  "SubSecCreateDate": "2015:03:28 23:45:05.967",
  "SubSecDateTimeOriginal": "2015:03:28 23:45:05.967",
  "ThumbnailImage": "(Binary data 10896 bytes, use -b option to extract)",
  "GPSAltitude": "34.5 m Above Sea Level",
  "GPSDateTime": "2015:03:28 20:44:45Z",
  "GPSLatitude": "52 deg 31' 24.98\" N",
  "GPSLongitude": "13 deg 23' 13.18\" E",
  "CircleOfConfusion": "0.004 mm",
  "FOV": "63.7 deg",
  "FocalLength35efl": "4.2 mm (35 mm equivalent: 29.0 mm)",
  "GPSPosition": "52 deg 31' 24.98\" N, 13 deg 23' 13.18\" E",
  "HyperfocalDistance": "1.82 m",
  "LightValue": 7.5
}]
```

As you can see UTC time (Zulu time) is:
`  "GPSDateTime": "2015:03:28 20:44:45Z"`

But after import UTC time is 22:45:05:
<img width="407" alt="Screenshot 2020-12-22 at 08 21 16" src="https://user-images.githubusercontent.com/64733669/102860735-b6251500-442e-11eb-8def-490f027b556f.png">
 
So it looks like iPhone photos also have incorrect UTC time, but correct local time.

**Comment by lastzero** [comment]
GPSDateTime can be wrong as well. Don't want to experiment too much with metadata shortly before the release. Also note I won't be able to code much more today, not at home for Christmas. Can we keep it like that for now?

**Comment by uxusoft** [comment]
Thank you so much for bug fix! If is much better because now it is possible to import huge collection.

I will take photos today, test and report here.

**Comment by uxusoft** [comment]
I made photo and video today to test:
Device is iPhone 8 iOS 13.3 (latest)
real local time (CET UTC+1) is ~9:41:

![IMG_250A5A0A2D30-1](https://user-images.githubusercontent.com/64733669/102870900-f8098780-443d-11eb-897e-d12ca727ee49.jpeg)

1. photo with GPS
2. photo without GPS
3. video with GPS
4. video without GPS

**1. [photo](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_6885.HEIC) with GPS**
```
[{
  "SourceFile": "IMG_6885.HEIC",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_6885.HEIC",
  "Directory": ".",
  "FileSize": "1190 kB",
  "FileModifyDate": "2020:12:22 09:41:54+01:00",
  "FileAccessDate": "2020:12:22 10:02:35+01:00",
  "FileInodeChangeDate": "2020:12:22 10:02:34+01:00",
  "FilePermissions": "rw-r--r--",
  "FileType": "HEIC",
  "FileTypeExtension": "heic",
  "MIMEType": "image/heic",
  "MajorBrand": "High Efficiency Image Format HEVC still image (.HEIC)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["mif1","MiPr","miaf","MiHB","heic"],
  "HandlerType": "Picture",
  "PrimaryItemReference": 49,
  "ExifByteOrder": "Big-endian (Motorola, MM)",
  "Make": "Apple",
  "Model": "iPhone 8",
  "Orientation": "Rotate 90 CW",
  "XResolution": 72,
  "YResolution": 72,
  "ResolutionUnit": "inches",
  "Software": 14.3,
  "ModifyDate": "2020:12:22 09:41:54",
  "HostComputer": "iPhone 8",
  "YCbCrPositioning": "Centered",
  "ExposureTime": "1/120",
  "FNumber": 1.8,
  "ExposureProgram": "Program AE",
  "ISO": 64,
  "ExifVersion": "0232",
  "DateTimeOriginal": "2020:12:22 09:41:54",
  "CreateDate": "2020:12:22 09:41:54",
  "OffsetTime": "+01:00",
  "OffsetTimeOriginal": "+01:00",
  "OffsetTimeDigitized": "+01:00",
  "ComponentsConfiguration": "Y, Cb, Cr, -",
  "ShutterSpeedValue": "1/120",
  "ApertureValue": 1.8,
  "BrightnessValue": 4.482354219,
  "ExposureCompensation": 0,
  "MeteringMode": "Multi-segment",
  "Flash": "Off, Did not fire",
  "FocalLength": "4.0 mm",
  "SubjectArea": "2015 1511 2217 1330",
  "RunTimeFlags": "Valid",
  "RunTimeValue": 107654508749375,
  "RunTimeScale": 1000000000,
  "RunTimeEpoch": 0,
  "AccelerationVector": "-0.03185654431 -0.9889873863 -0.1816732884",
  "HDRImageType": "Unknown (2)",
  "ContentIdentifier": "63BAB1BB-5984-4D8B-A189-A989D9427EEF",
  "SubSecTimeOriginal": 871,
  "SubSecTimeDigitized": 871,
  "FlashpixVersion": "0100",
  "ColorSpace": "Uncalibrated",
  "ExifImageWidth": 4032,
  "ExifImageHeight": 3024,
  "SensingMethod": "One-chip color area",
  "SceneType": "Directly photographed",
  "CustomRendered": "HDR (no original saved)",
  "ExposureMode": "Auto",
  "WhiteBalance": "Auto",
  "FocalLengthIn35mmFormat": "28 mm",
  "SceneCaptureType": "Standard",
  "LensInfo": "3.99mm f/1.8",
  "LensMake": "Apple",
  "LensModel": "iPhone 8 back camera 3.99mm f/1.8",
  "GPSLatitudeRef": "North",
  "GPSLongitudeRef": "East",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSSpeedRef": "km/h",
  "GPSSpeed": 14.90999985,
  "GPSImgDirectionRef": "Magnetic North",
  "GPSImgDirection": 195.6034393,
  "GPSDestBearingRef": "Magnetic North",
  "GPSDestBearing": 195.6034393,
  "GPSDateStamp": "2020:12:22",
  "GPSHPositioningError": "10 m",
  "ProfileCMMType": "Apple Computer Inc.",
  "ProfileVersion": "4.0.0",
  "ProfileClass": "Display Device Profile",
  "ColorSpaceData": "RGB ",
  "ProfileConnectionSpace": "XYZ ",
  "ProfileDateTime": "2017:07:07 13:22:32",
  "ProfileFileSignature": "acsp",
  "PrimaryPlatform": "Apple Computer Inc.",
  "CMMFlags": "Not Embedded, Independent",
  "DeviceManufacturer": "Apple Computer Inc.",
  "DeviceModel": "",
  "DeviceAttributes": "Reflective, Glossy, Positive, Color",
  "RenderingIntent": "Perceptual",
  "ConnectionSpaceIlluminant": "0.9642 1 0.82491",
  "ProfileCreator": "Apple Computer Inc.",
  "ProfileID": "ca1a9582257f104d389913d5d1ea1582",
  "ProfileDescription": "Display P3",
  "ProfileCopyright": "Copyright Apple Inc., 2017",
  "MediaWhitePoint": "0.95045 1 1.08905",
  "RedMatrixColumn": "0.51512 0.2412 -0.00105",
  "GreenMatrixColumn": "0.29198 0.69225 0.04189",
  "BlueMatrixColumn": "0.1571 0.06657 0.78407",
  "RedTRC": "(Binary data 32 bytes, use -b option to extract)",
  "ChromaticAdaptation": "1.04788 0.02292 -0.0502 0.02959 0.99048 -0.01706 -0.00923 0.01508 0.75168",
  "BlueTRC": "(Binary data 32 bytes, use -b option to extract)",
  "GreenTRC": "(Binary data 32 bytes, use -b option to extract)",
  "HEVCConfigurationVersion": 1,
  "GeneralProfileSpace": "Conforming",
  "GeneralTierFlag": "Main Tier",
  "GeneralProfileIDC": "Main Still Picture",
  "GenProfileCompatibilityFlags": "Main Still Picture, Main 10, Main",
  "ConstraintIndicatorFlags": "176 0 0 0 0 0",
  "GeneralLevelIDC": "90 (level 3.0)",
  "MinSpatialSegmentationIDC": 0,
  "ParallelismType": 0,
  "ChromaFormat": "4:2:0",
  "BitDepthLuma": 8,
  "BitDepthChroma": 8,
  "AverageFrameRate": 0,
  "ConstantFrameRate": "Unknown",
  "NumTemporalLayers": 1,
  "TemporalIDNested": "No",
  "ImageWidth": 4032,
  "ImageHeight": 3024,
  "ImageSpatialExtent": "4032x3024",
  "Rotation": 270,
  "ImagePixelDepth": "8 8 8",
  "MediaDataSize": 1215095,
  "MediaDataOffset": 3432,
  "RunTimeSincePowerUp": "1 days 5:54:15",
  "Aperture": 1.8,
  "ImageSize": "4032x3024",
  "LensID": "iPhone 8 back camera 3.99mm f/1.8",
  "Megapixels": 12.2,
  "ScaleFactor35efl": 7.0,
  "ShutterSpeed": "1/120",
  "SubSecCreateDate": "2020:12:22 09:41:54.871+01:00",
  "SubSecDateTimeOriginal": "2020:12:22 09:41:54.871+01:00",
  "SubSecModifyDate": "2020:12:22 09:41:54+01:00",
  "GPSAltitude": "426.7 m Above Sea Level",
  "GPSLatitude": "46 deg 11' 19.67\" N",
  "GPSLongitude": "6 deg 6' 49.49\" E",
  "CircleOfConfusion": "0.004 mm",
  "FOV": "65.5 deg",
  "FocalLength35efl": "4.0 mm (35 mm equivalent: 28.0 mm)",
  "GPSPosition": "46 deg 11' 19.67\" N, 6 deg 6' 49.49\" E",
  "HyperfocalDistance": "2.07 m",
  "LightValue": 9.2
}]
```
<img width="435" alt="Screenshot 2020-12-22 at 10 12 18" src="https://user-images.githubusercontent.com/64733669/102871073-40c14080-443e-11eb-909f-a294eef48e74.png">

Local/UTC is correct!

**2. [photo](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_6885_nogps.HEIC) without GPS**
```
[{
  "SourceFile": "IMG_6885_nogps.HEIC",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_6885_nogps.HEIC",
  "Directory": ".",
  "FileSize": "1174 kB",
  "FileModifyDate": "2020:12:22 10:02:50+01:00",
  "FileAccessDate": "2020:12:22 10:03:19+01:00",
  "FileInodeChangeDate": "2020:12:22 10:03:18+01:00",
  "FilePermissions": "rw-r--r--",
  "FileType": "HEIC",
  "FileTypeExtension": "heic",
  "MIMEType": "image/heic",
  "MajorBrand": "High Efficiency Image Format HEVC still image (.HEIC)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["mif1","miaf","MiHB","heic"],
  "HandlerType": "Picture",
  "PrimaryItemReference": 49,
  "ExifByteOrder": "Big-endian (Motorola, MM)",
  "Make": "Apple",
  "Model": "iPhone 8",
  "Orientation": "Rotate 90 CW",
  "XResolution": 72,
  "YResolution": 72,
  "ResolutionUnit": "inches",
  "Software": 14.3,
  "HostComputer": "iPhone 8",
  "TileWidth": 512,
  "TileLength": 512,
  "ExposureTime": "1/120",
  "FNumber": 1.8,
  "ExposureProgram": "Program AE",
  "ISO": 64,
  "ExifVersion": "0232",
  "DateTimeOriginal": "2020:12:22 09:41:54",
  "OffsetTime": "+01:00",
  "OffsetTimeOriginal": "+01:00",
  "OffsetTimeDigitized": "+01:00",
  "ComponentsConfiguration": "Y, Cb, Cr, -",
  "ShutterSpeedValue": "1/120",
  "ApertureValue": 1.8,
  "BrightnessValue": 4.482354219,
  "ExposureCompensation": 0,
  "MeteringMode": "Multi-segment",
  "Flash": "Off, Did not fire",
  "FocalLength": "4.0 mm",
  "SubjectArea": "2015 1511 2217 1330",
  "RunTimeFlags": "Valid",
  "RunTimeValue": 107654508749375,
  "RunTimeScale": 1000000000,
  "RunTimeEpoch": 0,
  "AccelerationVector": "-0.03185654431 -0.9889873863 -0.1816732884",
  "HDRImageType": "Unknown (2)",
  "ContentIdentifier": "FC6EB877-7AA2-4B22-B2B0-C6EE13BA2657",
  "SubSecTimeOriginal": 871,
  "SubSecTimeDigitized": 871,
  "FlashpixVersion": "0100",
  "ColorSpace": "Uncalibrated",
  "ExifImageWidth": 4032,
  "ExifImageHeight": 3024,
  "SensingMethod": "One-chip color area",
  "SceneType": "Directly photographed",
  "CustomRendered": "HDR (no original saved)",
  "ExposureMode": "Auto",
  "WhiteBalance": "Auto",
  "FocalLengthIn35mmFormat": "28 mm",
  "SceneCaptureType": "Standard",
  "LensInfo": "3.99000001mm f/1.8",
  "LensMake": "Apple",
  "LensModel": "iPhone 8 back camera 3.99mm f/1.8",
  "XMPToolkit": "XMP Core 6.0.0",
  "CreateDate": "2020:12:22 09:41:54",
  "CreatorTool": 14.3,
  "ModifyDate": "2020:12:22 09:41:54",
  "DateCreated": "2020:12:22 09:41:54",
  "ProfileCMMType": "Apple Computer Inc.",
  "ProfileVersion": "4.0.0",
  "ProfileClass": "Display Device Profile",
  "ColorSpaceData": "RGB ",
  "ProfileConnectionSpace": "XYZ ",
  "ProfileDateTime": "2017:07:07 13:22:32",
  "ProfileFileSignature": "acsp",
  "PrimaryPlatform": "Apple Computer Inc.",
  "CMMFlags": "Not Embedded, Independent",
  "DeviceManufacturer": "Apple Computer Inc.",
  "DeviceModel": "",
  "DeviceAttributes": "Reflective, Glossy, Positive, Color",
  "RenderingIntent": "Perceptual",
  "ConnectionSpaceIlluminant": "0.9642 1 0.82491",
  "ProfileCreator": "Apple Computer Inc.",
  "ProfileID": "ca1a9582257f104d389913d5d1ea1582",
  "ProfileDescription": "Display P3",
  "ProfileCopyright": "Copyright Apple Inc., 2017",
  "MediaWhitePoint": "0.95045 1 1.08905",
  "RedMatrixColumn": "0.51512 0.2412 -0.00105",
  "GreenMatrixColumn": "0.29198 0.69225 0.04189",
  "BlueMatrixColumn": "0.1571 0.06657 0.78407",
  "RedTRC": "(Binary data 32 bytes, use -b option to extract)",
  "ChromaticAdaptation": "1.04788 0.02292 -0.0502 0.02959 0.99048 -0.01706 -0.00923 0.01508 0.75168",
  "BlueTRC": "(Binary data 32 bytes, use -b option to extract)",
  "GreenTRC": "(Binary data 32 bytes, use -b option to extract)",
  "HEVCConfigurationVersion": 1,
  "GeneralProfileSpace": "Conforming",
  "GeneralTierFlag": "Main Tier",
  "GeneralProfileIDC": "Main Still Picture",
  "GenProfileCompatibilityFlags": "Main Still Picture, Main 10, Main",
  "ConstraintIndicatorFlags": "176 0 0 0 0 0",
  "GeneralLevelIDC": "90 (level 3.0)",
  "MinSpatialSegmentationIDC": 0,
  "ParallelismType": 0,
  "ChromaFormat": "4:2:0",
  "BitDepthLuma": 8,
  "BitDepthChroma": 8,
  "AverageFrameRate": 0,
  "ConstantFrameRate": "Unknown",
  "NumTemporalLayers": 1,
  "TemporalIDNested": "No",
  "ImageWidth": 4032,
  "ImageHeight": 3024,
  "ImageSpatialExtent": "4032x3024",
  "Rotation": 270,
  "ImagePixelDepth": "8 8 8",
  "MediaDataSize": 1199127,
  "MediaDataOffset": 3309,
  "RunTimeSincePowerUp": "1 days 5:54:15",
  "Aperture": 1.8,
  "ImageSize": "4032x3024",
  "LensID": "iPhone 8 back camera 3.99mm f/1.8",
  "Megapixels": 12.2,
  "ScaleFactor35efl": 7.0,
  "ShutterSpeed": "1/120",
  "SubSecCreateDate": "2020:12:22 09:41:54.871+01:00",
  "SubSecDateTimeOriginal": "2020:12:22 09:41:54.871+01:00",
  "SubSecModifyDate": "2020:12:22 09:41:54+01:00",
  "CircleOfConfusion": "0.004 mm",
  "FOV": "65.5 deg",
  "FocalLength35efl": "4.0 mm (35 mm equivalent: 28.0 mm)",
  "HyperfocalDistance": "2.07 m",
  "LightValue": 9.2
}]
```

<img width="382" alt="Screenshot 2020-12-22 at 10 18 35" src="https://user-images.githubusercontent.com/64733669/102871752-1623b780-443f-11eb-8a94-cbdfc9cf2138.png">

**UTC time is not correct! local != UTC**

**3. [video](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_6886.MOV) with GPS**
```
[{
  "SourceFile": "IMG_6886.MOV",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_6886.MOV",
  "Directory": ".",
  "FileSize": "7.7 MB",
  "FileModifyDate": "2020:12:22 09:41:58+01:00",
  "FileAccessDate": "2020:12:22 09:41:58+01:00",
  "FileInodeChangeDate": "2020:12:22 10:02:34+01:00",
  "FilePermissions": "rw-r--r--",
  "FileType": "MOV",
  "FileTypeExtension": "mov",
  "MIMEType": "video/quicktime",
  "MajorBrand": "Apple QuickTime (.MOV/QT)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["qt  "],
  "MediaDataSize": 8110211,
  "MediaDataOffset": 36,
  "MovieHeaderVersion": 0,
  "CreateDate": "2020:12:22 08:41:58",
  "ModifyDate": "2020:12:22 08:42:04",
  "TimeScale": 600,
  "Duration": "5.48 s",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 6,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2020:12:22 08:41:58",
  "TrackModifyDate": "2020:12:22 08:42:04",
  "TrackID": 1,
  "TrackDuration": "5.48 s",
  "TrackLayer": 0,
  "TrackVolume": "0.00%",
  "ImageWidth": 1920,
  "ImageHeight": 1080,
  "CleanApertureDimensions": "1920x1080",
  "ProductionApertureDimensions": "1920x1080",
  "EncodedPixelsDimensions": "1920x1080",
  "GraphicsMode": "ditherCopy",
  "OpColor": "32768 32768 32768",
  "CompressorID": "hvc1",
  "SourceImageWidth": 1920,
  "SourceImageHeight": 1080,
  "XResolution": 72,
  "YResolution": 72,
  "CompressorName": "HEVC",
  "BitDepth": 24,
  "VideoFrameRate": 60,
  "Balance": 0,
  "AudioFormat": "mp4a",
  "AudioChannels": 1,
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "PurchaseFileFormat": "mp4a",
  "Warning": "[minor] The ExtractEmbedded option may find more tags in the media data",
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "ContentDescribes": "Track 1",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2020:12:22 08:41:58",
  "MediaModifyDate": "2020:12:22 08:42:04",
  "MediaTimeScale": 600,
  "MediaDuration": "5.48 s",
  "MediaLanguageCode": "und",
  "GenMediaVersion": 0,
  "GenFlags": "0 0 0",
  "GenGraphicsMode": "ditherCopy",
  "GenOpColor": "32768 32768 32768",
  "GenBalance": 0,
  "HandlerClass": "Data Handler",
  "HandlerVendorID": "Apple",
  "HandlerDescription": "Core Media Data Handler",
  "MetaFormat": "mebx",
  "HandlerType": "Metadata Tags",
  "LocationAccuracyHorizontal": 10.000000,
  "GPSCoordinates": "46 deg 11' 21.12\" N, 6 deg 6' 47.16\" E, 420.771 m Above Sea Level",
  "Make": "Apple",
  "Model": "iPhone 8",
  "Software": 14.3,
  "CreationDate": "2020:12:22 09:41:58+01:00",
  "ImageSize": "1920x1080",
  "Megapixels": 2.1,
  "AvgBitrate": "11.8 Mbps",
  "GPSAltitude": "420.771 m",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSLatitude": "46 deg 11' 21.12\" N",
  "GPSLongitude": "6 deg 6' 47.16\" E",
  "Rotation": 90,
  "GPSPosition": "46 deg 11' 21.12\" N, 6 deg 6' 47.16\" E"
}]
```
<img width="381" alt="Screenshot 2020-12-22 at 10 15 06" src="https://user-images.githubusercontent.com/64733669/102871331-9695e880-443e-11eb-9d7f-3a44675b41e0.png">

Local/UTC is correct!

**4. [video](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_6886_nogps.MOV) without GPS**
```
[{
  "SourceFile": "IMG_6886_nogps.MOV",
  "ExifToolVersion": 12.00,
  "FileName": "IMG_6886_nogps.MOV",
  "Directory": ".",
  "FileSize": "7.7 MB",
  "FileModifyDate": "2020:12:22 10:02:49+01:00",
  "FileAccessDate": "2020:12:22 10:02:49+01:00",
  "FileInodeChangeDate": "2020:12:22 10:03:24+01:00",
  "FilePermissions": "rw-r--r--",
  "FileType": "MOV",
  "FileTypeExtension": "mov",
  "MIMEType": "video/quicktime",
  "MajorBrand": "Apple QuickTime (.MOV/QT)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["qt  "],
  "MediaDataSize": 8062831,
  "MediaDataOffset": 36,
  "MovieHeaderVersion": 0,
  "CreateDate": "2020:12:22 09:02:49",
  "ModifyDate": "2020:12:22 09:02:49",
  "TimeScale": 600,
  "Duration": "5.48 s",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 5,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2020:12:22 09:02:49",
  "TrackModifyDate": "2020:12:22 09:02:49",
  "TrackID": 1,
  "TrackDuration": "5.48 s",
  "TrackLayer": 0,
  "TrackVolume": "0.00%",
  "ImageWidth": 1920,
  "ImageHeight": 1080,
  "CleanApertureDimensions": "1920x1080",
  "ProductionApertureDimensions": "1920x1080",
  "EncodedPixelsDimensions": "1920x1080",
  "GraphicsMode": "ditherCopy",
  "OpColor": "32768 32768 32768",
  "CompressorID": "hvc1",
  "SourceImageWidth": 1920,
  "SourceImageHeight": 1080,
  "XResolution": 72,
  "YResolution": 72,
  "CompressorName": "HEVC",
  "BitDepth": 24,
  "VideoFrameRate": 60,
  "Balance": 0,
  "AudioFormat": "mp4a",
  "AudioChannels": 1,
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "PurchaseFileFormat": "mp4a",
  "Warning": "[minor] The ExtractEmbedded option may find more tags in the media data",
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "ContentDescribes": "Track 1",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2020:12:22 09:02:49",
  "MediaModifyDate": "2020:12:22 09:02:49",
  "MediaTimeScale": 600,
  "MediaDuration": "5.48 s",
  "MediaLanguageCode": "und",
  "GenMediaVersion": 0,
  "GenFlags": "0 0 0",
  "GenGraphicsMode": "ditherCopy",
  "GenOpColor": "32768 32768 32768",
  "GenBalance": 0,
  "HandlerClass": "Data Handler",
  "HandlerVendorID": "Apple",
  "HandlerDescription": "Core Media Data Handler",
  "MetaFormat": "mebx",
  "HandlerType": "Metadata Tags",
  "LocationAccuracyHorizontal": 10.000000,
  "Make": "Apple",
  "Model": "iPhone 8",
  "Software": 14.3,
  "CreationDate": "2020:12:22 09:41:58+01:00",
  "ImageSize": "1920x1080",
  "Megapixels": 2.1,
  "AvgBitrate": "11.8 Mbps",
  "Rotation": 90
}]
```

<img width="369" alt="Screenshot 2020-12-22 at 10 21 54" src="https://user-images.githubusercontent.com/64733669/102872095-8d594b80-443f-11eb-9b59-12787b242464.png">

**UTC time is not correct! local != UTC**

==
I think it is more than enough and timezones are fixed now. It may be good to add support for CreationDate for Apple devices (time and timezone) in future.

Thank you very much!

**Comment by lastzero** [comment]
We set local = UTC when there is no time zone. It changes when you set coordinates. Might be a bit misleading. In fact, we should hide the UTC time in this case to avoid confusion as there only is a local time.

**Comment by uxusoft** [comment]
Yes, you are right. I think bug is fully fixed.

**Comment by lastzero** [comment]
Just tested the dates you mentioned in the chat:

```
"MediaCreateDate": "2016:10:25 09:53:58", 
"CreationDate": "2016:10:25 12:53:58+03:00".
```

Result:

Local Time: 2016-10-25 12:53:58
UTC: 2016-10-25 09:53:58

Is it possible that the JSON file additionally contains DateTimeOriginal without a time zone? This is typically not UTC. Had to manually create a JSON file for testing because you didn't give me any more data to work with.

Please create a new issue with an example file for testing and let us know how important this is.
